pydantic/annotated_types.py

```diff
@@ -0,0 +1,59 @@
+from typing import TYPE_CHECKING, Any, Dict, FrozenSet, NamedTuple, Type
+
+from .fields import Required
+from .main import BaseModel, create_model
+
+if TYPE_CHECKING:
+
+    class TypedDict(Dict[str, Any]):
```

Comment:
We could use `typing_extensions.TypedDict` for python < 3.9 or `typing.TypedDict` for 3.9+ but we would need to add `typing_extensions` in linting requirements.